tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -122,24 +123,43 @@ impl LeaderTpuCache {
         )
     }
 
-    // Get the TPU sockets for the current leader and upcoming leaders according to fanout size
+    // Get the TPU sockets for the current leader and upcoming *unique* leaders according to fanout size.
+    fn get_unique_leader_sockets(
+        &self,
+        estimated_current_slot: Slot,
+        fanout_slots: u64,
+    ) -> Vec<SocketAddr> {
+        let all_leader_sockets = self.get_leader_sockets(estimated_current_slot, fanout_slots);
+
+        let mut unique_sockets = Vec::new();
+        let mut seen = HashSet::new();
```

Comment:
@gregcusack how do you fell about this code? One more allocation in comparison to the original code, it is needed to reuse `get_leader_sockets` instead of just leaving old code as it is.

Reply:
how many slots ahead does it get the leader schedule? can't imagine the `seen` gets that big